assembly/src/ast/error_msg.rs

```diff
@@ -0,0 +1,32 @@
+use alloc::string::{String, ToString};
+
+use vm_core::crypto::hash::{Blake3_256, Blake3Digest, Digest};
+
+use super::Immediate;
+use crate::Felt;
+
+/// The type of error messages used in MASM assertions.
+pub type ErrorMsg = Immediate<String>;
+
+/// MAST error codes.
+pub type ErrorCode = Felt;
+
+// This function is mostly for testing. Use `from` instead.
+pub fn error_code_from_msg(msg: String) -> u64 {
+    let digest: Blake3Digest<32> = Blake3_256::hash(msg.as_bytes());
+    let mut digest_bytes: [u8; 8] = [0; 8];
+    digest_bytes.copy_from_slice(&digest.as_bytes()[0..8]);
+    u64::from_le_bytes(digest_bytes)
+}
+
+impl From<&ErrorMsg> for u64 {
```

Comment:
This seems to indicate that `ErrorMsg` should never be a constant name, so I'd definitely make the value type `Ident` or `Arc<str>` instead.